packages/earth-map/src/pages/verify-email/VerifyEmail.tsx

```diff
@@ -28,41 +29,49 @@ import './styles.scss';
 
 export const APP_LOGO = require('images/unbl_logo.svg');
 
-interface EmailState {
-  type: 'inactive' | 'pending' | 'done' | 'error';
+enum EmailStates {
+  INACTIVE = 'Inactive',
+  PENDING = 'Pending',
+  DONE = 'Done',
+  ERROR = 'Error',
+}
+
+interface VerifyEmailState {
+  type: EmailStates;
   text: string;
+  message?: string;
 }
 
-const VerifyEmail = ({ returnToHome }) => {
+const VerifyEmail = () => {
   const { logout, isEmailVerified } = useAuth0();
 
   return (
     <div className="verified-page marapp-qa-verify-email">
       <div className="verified-container">
-        {isEmailVerified ? (
-          <VerifiedEmailTemplate returnToHome={returnToHome} />
-        ) : (
-          <VerifyEmailTemplate logout={logout} />
-        )}
+        {isEmailVerified ? <VerifiedEmailTemplate /> : <VerifyEmailTemplate logout={logout} />}
       </div>
     </div>
   );
 };
 
 const VerifyEmailTemplate = ({ logout }) => {
-  const [emailState, setEmailState] = useState<EmailState>({
-    type: 'inactive',
+  const [emailState, setEmailState] = useState<VerifyEmailState>({
+    type: EmailStates.INACTIVE,
     text: 'Send email verification Link',
   });
 
   const handleResendEmail = async () => {
-    setEmailState({ type: 'pending', text: 'sending email validation' });
+    setEmailState({ type: EmailStates.PENDING, text: 'sending email validation' });
 
     try {
       await ProfileService.resendEmailConfirmation();
-      setEmailState({ type: 'done', text: 'email sent' });
+      setEmailState({ type: EmailStates.DONE, text: 'email sent' });
     } catch (error) {
-      setEmailState({ type: 'error', text: 'error' });
+      setEmailState({
+        type: EmailStates.ERROR,
+        text: 'something went wrong',
+        message: error.data.errors[0].detail,
```

Comment:
I think `error` is undefined if network connection fails.  Please handle that case too.